cmd/bucket-replication.go

```diff
@@ -1706,6 +1730,7 @@ type ReplicationPool struct {
 	mrfSaveCh       chan MRFReplicateEntry
 	mrfStopCh       chan struct{}
 	mrfWorkerSize   int
+	mrfEntries      map[string]MRFReplicateEntry
```

Comment:
so this is going to be a million entries worst case? @poornas

Reply:
Having this as map I am not sure is a good idea.

Reply:
there would be duplicates if it were not a map - surely 1M shouldn't be that big of a deal?

Reply:
it is a big deal @poornas - map and 1m entry lookups are huge, here the values are fat as well since its a data struct value of MRFReplicateEntry

Reply:
I don't think this PR makes sense to add it will end up causing more memory issues AFAICS.

Reply:
you are misunderstanding - this is not a new map, just the MRF map moved into ReplicationPool struct so we know how many entries are there in the map early on. If we remove this map, there will be no MRF queueing to speak of.